flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -1477,6 +1477,31 @@
                     .outputTypeStrategy(TypeStrategies.MISSING)
                     .build();
 
+    // --------------------------------------------------------------------------------------------
+    // JSON functions
+    // --------------------------------------------------------------------------------------------
+
+    public static final BuiltInFunctionDefinition JSON_EXISTS =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("JSON_EXISTS")
+                    .kind(SCALAR)
+                    .inputTypeStrategy(
+                            or(
+                                    sequence(
+                                            logical(LogicalTypeFamily.CHARACTER_STRING),
+                                            and(
+                                                    logical(LogicalTypeFamily.CHARACTER_STRING),
+                                                    LITERAL)),
+                                    sequence(
+                                            logical(LogicalTypeFamily.CHARACTER_STRING),
+                                            and(
+                                                    logical(LogicalTypeFamily.CHARACTER_STRING),
+                                                    LITERAL),
+                                            logical(LogicalTypeRoot.SYMBOL))))
+                    .outputTypeStrategy(nullable(explicit(DataTypes.BOOLEAN())))
```

Comment:
this is not correct, even when both args are NOT NULL the function can still return NULL depending on the JsonExistsOnError strategy